src/snowflake/snowpark/session.py

```diff
@@ -1856,18 +1858,34 @@ def sproc(self) -> StoredProcedureRegistration:
         """
         return self._sp_registration
 
+    def _infer_is_return_table(self, sproc_name: str, *args: Any) -> bool:
+        try:
+            arg_types = [convert_sp_to_sf_type(infer_type(arg)) for arg in args]
+            func_signature = f"{sproc_name.upper()}({', '.join(arg_types)})"
+            sproc_desc = self._run_query(f"describe procedure {func_signature}")
+            return_type = sproc_desc[1][1]
```

Comment:
Let's add a comment about the returned describe result so it is easy to understand this result extraction.

Reply:
Added comment about return type and addressed Yijun's `DataFrame` comments.